core/src/test/java/org/kiwix/kiwixmobile/core/history/viewmodel/HistoryViewModelTest.kt

```diff
@@ -0,0 +1,474 @@
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.kiwix.kiwixmobile.core.base.SideEffect
+import org.kiwix.kiwixmobile.core.dao.HistoryDao
+import org.kiwix.kiwixmobile.core.history.adapter.HistoryListItem
+import org.kiwix.kiwixmobile.core.history.adapter.HistoryListItem.DateItem
+import org.kiwix.kiwixmobile.core.history.adapter.HistoryListItem.HistoryItem
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action.ExitActionModeMenu
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action.ExitHistory
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action.Filter
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action.OnItemClick
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action.OnItemLongClick
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action.UpdateHistory
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action.UserClickedConfirmDelete
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action.UserClickedDeleteButton
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action.UserClickedDeleteSelectedHistoryItems
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action.UserClickedShowAllToggle
+import org.kiwix.kiwixmobile.core.history.viewmodel.effects.DeleteHistoryItems
+import org.kiwix.kiwixmobile.core.history.viewmodel.effects.OpenHistoryItem
+import org.kiwix.kiwixmobile.core.history.viewmodel.effects.ShowDeleteHistoryDialog
+import org.kiwix.kiwixmobile.core.history.viewmodel.effects.UpdateAllHistoryPreference
+import org.kiwix.kiwixmobile.core.reader.ZimReaderContainer
+import org.kiwix.kiwixmobile.core.search.viewmodel.effects.Finish
+import org.kiwix.kiwixmobile.core.utils.SharedPreferenceUtil
+import org.kiwix.sharedFunctions.InstantExecutorExtension
+import org.kiwix.sharedFunctions.setScheduler
+
+@ExtendWith(InstantExecutorExtension::class)
+internal class HistoryViewModelTest {
+  private val historyDao: HistoryDao = mockk()
+  private val zimReaderContainer: ZimReaderContainer = mockk()
+  private val sharedPreferenceUtil: SharedPreferenceUtil = mockk()
+
+  lateinit var viewModel: HistoryViewModel
+  private val testScheduler = TestScheduler()
+
+  init {
+    setScheduler(testScheduler)
+    RxJavaPlugins.setIoSchedulerHandler { Schedulers.trampoline() }
+  }
+
+  private val itemsFromDb: PublishProcessor<List<HistoryItem>> =
+    PublishProcessor.create()
+
+  @BeforeEach
+  fun init() {
+    clearAllMocks()
+    every { zimReaderContainer.id } returns "id"
+    every { zimReaderContainer.name } returns "zimName"
+    every { sharedPreferenceUtil.showHistoryAllBooks } returns true
+    every { historyDao.history() } returns itemsFromDb
+    viewModel = HistoryViewModel(historyDao, zimReaderContainer, sharedPreferenceUtil)
+  }
+
+  private fun resultsIn(st: HistoryState) {
+    viewModel.state.test().assertValue(st)
+  }
+
+  @Nested
```

Comment:
We probably don't even need for this to be a nested class. You can see the benefits of nested classes from [this article](https://phauer.com/2018/best-practices-unit-testing-kotlin/) that I based our test setup on, essentially the class HistoryViewModel is so homogenous that there isn't a need for a split